lib/NetworkManager.js

```diff
@@ -249,7 +250,8 @@ class NetworkManager extends EventEmitter {
         redirectChain = request._redirectChain;
       }
     }
-    this._handleRequestStart(event.requestId, null, event.request.url, event.type, event.request, event.frameId, redirectChain);
+    const isNavigationRequest = event.requestId === event.loaderId && event.type === 'Document';
```

Comment:
The requestId can’t equal the loaderId and it not be a document request right? What happens when navigating to an image?

Reply:
The `request.resourceType()` is inferred by the request initiator rather then URL extension; so yes, it should always be `document` as far as I know. 

I'll add a test for navigation to the image to prove the point.